Source/HTTP/httpcall_request.cpp

```diff
@@ -6,6 +6,35 @@
 
 using namespace xbox::httpclient;
 
+HRESULT CALLBACK DefaultRequestBodyReadFunction(
```

Comment:
The request body read and response body write callbacks can be invoked repeatedly until all data has been sent or received. Each invocation reads/writes the next chunk to be processed and in the case of the read callback is tracked by the “offset” argument to the callback.

As for JNI round-trips, on average we shouldn’t have additional JNI calls since OkHttp sends data in 8KiB chunks and most requests are much smaller than that. But in the case of larger requests, we’ll incur the cost of additional JNI calls. However we reduce our memory usage by one half since we avoid duplicating the body’s data in the JVM and let OkHttp read our native buffers (or files) directly in 8KiB blocks

Reply:
Sounds good. I think I had misunderstood something and was concerned that there would now be an additional JNI round-trip even if there was only one chunk, but I don't think that's necessarily the case. Regardless, if it's 8kb chunks, yeah not worried.